core/src/main/java/org/apache/accumulo/core/data/ResourceRequest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.data;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+
+import org.apache.accumulo.core.data.thrift.TResourceRequest;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+import org.apache.hadoop.io.WritableComparable;
+
+/**
+ * Defines a request for resources.
+ *
+ */
+public class ResourceRequest implements WritableComparable<ResourceRequest>, Cloneable {
```

Comment:
Would be nice to not make this API Pojo implement WritableComparable.  Including other library APIs in Accumulo's API negatively impacts our ability to evolve and control the Accumulo API over time.   Also how its serialized is an implementation detail.  We can have non public util code that serializes this Pojo.   I know Accumulo APIs currently do this, but those are mistakes we made in the past.  We can avoid making the same mistakes for new APIs.